pybossa/cache/projects.py

```diff
@@ -38,16 +38,14 @@
 @cache(timeout=timeouts.get('STATS_FRONTPAGE_TIMEOUT'),
        key_prefix="front_page_top_projects")
 def get_top(n=4):
-    """Return top n=4 projects."""
-    sql = text('''SELECT project.id, project.name, project.short_name, project.description,
-               project.info,
-               COUNT(project_id) AS total
-               FROM task_run, project
-               WHERE project_id IS NOT NULL
-               AND project.id=project_id
-               AND project.published=True
-               AND (project.info->>'passwd_hash') IS NULL
-               GROUP BY project.id ORDER BY total DESC LIMIT :limit;''')
+    """Return top n=4 projects with most task runs."""
+    sql = text('''SELECT p.id, p.name, p.short_name, p.description, p.info, ps.n_task_runs AS total
+                FROM project p
+                JOIN project_stats ps
```

Comment:
This is a good trade off for real time data vs speed. The original query was very slow and it always return empty row. Besides the top N task runs in projects shouldn't change a lot. We do get it refreshed every 10min and every 1hour('high' and 'super') [here](https://github.com/bloomberg/pybossa/blob/4b23a14f16c978181fcd79139d58528b3260e626/pybossa/jobs.py#L231).

Reply:
The earlier query has bug that got introduced with all project settings require password. From function name and docstring, the original goal of this function is to generate top 4 projects historically and not limited to last 3 months of data that would happen with using project_stats instead. Hence there are two possibilities here,
1. Confirm with team/project owner the expected behavior here. If historic data is expected, tune original query for performance improvement.
2. Update docstring and product documentation on expected behavior; this generates recent top 4 project stats instead of historically top 4 projects.

Reply:
will update the doc string